projects/memapCore/src/main/java/memap/controller/GuiController.java

```diff
@@ -188,7 +199,7 @@ public BuildingController deserialize(JsonElement jsonElement, Type typeOfT, Jso
 			JsonObject jObject = (JsonObject) jsonElement;
 
 			// Creating the building
-			BuildingController building = new CSVBuildingController(jObject.get("name").getAsString());
+			BuildingController building = new CSVBuildingController(jObject.get("formattedName").getAsString());

```

Comment:
quite dangerous change... every building / device that we have now, has no formattedName value... So loading it will fail. Failing leads to a none catched error... 

The same issue is also below. It somehow also transfers to showing end results...

Reply:
Several null-pointer checks now and different exception handlings are now available.